datafusion/common/src/scalar.rs

```diff
@@ -1112,14 +1117,14 @@ impl ScalarValue {
         scalars: impl IntoIterator<Item = ScalarValue>,
         precision: &usize,
         scale: &usize,
-    ) -> Result<DecimalArray> {
+    ) -> Result<Decimal128Array> {
```

Comment:
`DecimalArray` now is `DecimalArray128` to add support of `DecimalArray256` -- in https://github.com/apache/arrow-rs/issues/2101